### PR TITLE
Assert-Module: Reduce verbose output

### LINKED ISSUE #66: `Assert-Module:` Outputs unnecessary verbose statements due to `-ListAvailable` is used each time
#### Details of the scenario you tried and the problem that is occurring
Every time when running `Assert-Module` and `$VerbosePreference` is set to `'Continue'` the command outputs a lot of verbose statements depending on what module is asserted.

#### Steps to reproduce the problem

```powershell
$VerbosePreference = 'Continue'
Assert-Module -ModuleName 'DnsServer'
```

Outputs

```plaintext
VERBOSE: Loading module from path
'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\DnsServer\PS_DnsServerCache_v1.0.0.cdxml'.
VERBOSE: Loading module from path
'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\DnsServer\PS_DnsServerConditionalForwarder_v1.0.0.cdxml'.
VERBOSE: Loading module from path
'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\DnsServer\PS_DnsServerDiagnostics_v1.0.0.cdxml'.
VERBOSE: Loading module from path
'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\DnsServer\PS_DnsServerDirectoryPartition_v1.0.0.cdxml'.
VERBOSE: Loading module from path
'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\DnsServer\PS_DnsServerDsSetting_v1.0.0.cdxml'.
VERBOSE: Loading module from path

.... Removed rows for readability

VERBOSE: Loading module from path
'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\DnsServer\PS_DnsServerRecursionScope_v1.0.0.cdxml'.
VERBOSE: Loading module from path
'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\DnsServer\PS_DnsServerQueryResolutionPolicy_v1.0.0.cdxml'.
VERBOSE: Loading module from path
'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\DnsServer\PS_DnsServerZoneTransferPolicy_v1.0.0.cdxml'.
VERBOSE: Loading module from path
'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\DnsServer\PS_DnsServerPolicy_v1.0.0.cdxml'.
VERBOSE: Loading module from path
'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\DnsServer\PS_DnsServerResponseRateLimiting_v1.0.0.cdxml'.
VERBOSE: Loading module from path
'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\DnsServer\PS_DnsServerResponseRateLimitingExceptionlist_v1.0.0.cdxml'.
VERBOSE: Loading module from path
'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\DnsServer\PS_DnsServerVirtualizationInstance_v1.0.0.cdxml'.
```

#### Expected behavior
Only show these verbose messages when the module to assert is not in the session.

#### Current behavior
Always shows the verbose messages even if the module is already present in the session

#### Suggested solution to the issue

This is due to `-ListAvailable` is always used.

https://github.com/dsccommunity/DscResource.Common/blob/80005418c1457f75fb681222efde19299268acb6/source/Public/Assert-Module.ps1#L33

Suggest adding an evaluation `if (-not (Get-Module -Name $ModuleName))` before using `-ListAvailable`. Using that will not output the verbose messages,.

#### The operating system the target node is running
Windows Server 2016

#### Version and build of PowerShell the target node is running
5.1

#### Version of the module that was used
0.10.1

### PR DESCRIPTION
**Pull request by johlju**
#### Pull Request (PR) description
- `Assert-Module`
  - Now it possible to forcibly import a module using `-ImportModule -Force`
  - It no longer outputs verbose messages that is normally generated when
    using `Get-Module -ListAvailable` if the module that is asserted is
    already in the session (issue #66).

#### This Pull Request (PR) fixes the following issues

- Fixes #66

#### Task list

- [x] Added an entry to the change log under the Unreleased section of the
      file CHANGELOG.md. Entry should say what was changed and how that
      affects users (if applicable), and reference the issue being resolved
      (if applicable).
- [x] Documentation added/updated in README.md.
- [x] Comment-based help added/updated for all new/changed functions.
- [ ] Localization strings added/updated in all localization files as appropriate.
- [x] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] Integration tests added/updated (where possible). See
  [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

---
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/dscresource.common/67)

### CODE REVIEW

**Comment by johlju** [comment]
@PlagueHO when you have time can you review this one. There is no hurry.

File: source/Public/Assert-Module.ps1 [hunk 1]
```diff
@@ -9,12 +9,29 @@
         Specifies the name of the module to assert.
 
     .PARAMETER ImportModule
-        Specfiies to import the module if it is asserted.
+        Specifies to import the module if it is asserted.
+
+    .PARAMETER Force
+        Specifies to forcibly import the module even if it is already in the
+        session. THis parameter is ignored unless parameter `ImportModule` is
```
Comment (gaelcolas) [inline source/Public/Assert-Module.ps1:16]: non-blocking: This not THis

**Review by gaelcolas — APPROVED** [review]
Only found one typo. Feel free to merge.

File: source/Public/Assert-Module.ps1 (hunk 1 above)
Comment (johlju) [inline source/Public/Assert-Module.ps1:16]:
```suggestion
        session. THis parameter is ignored unless parameter `ImportModule` is
```
```suggestion
        session. This parameter is ignored unless parameter `ImportModule` is
```

**Pull request merged**